### PR TITLE
Conditionalized container vars in Makefile

### PR DESCRIPTION
**Pull request by dofinn**
Added the ability to use docker or podman along with setting a desired
tag.

Service account for observatorium-operator was updated to default to
observatorium namespace.

### CODE REVIEW

**Review by periklis — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,10 @@ BUILD_DATE := $(shell date -u +"%Y-%m-%d")
 BUILD_TIMESTAMP := $(shell date -u +"%Y-%m-%dT%H:%M:%S%Z")
 VCS_BRANCH := $(strip $(shell git rev-parse --abbrev-ref HEAD))
 VCS_REF := $(strip $(shell [ -d .git ] && git rev-parse --short HEAD))
-DOCKER_REPO ?= quay.io/observatorium/observatorium-operator
+CONTAINER_REPO ?= quay.io/observatorium/observatorium-operator
+CONTAINER_ENGINE ?= docker
```
Comment (jharrington22) [inline Makefile:10]: @dofinn you could change this too `CONTAINER_ENGINE=$(shell command -v podman 2>/dev/null || command -v docker 2>/dev/null)` to auto select the container engine like we do for our builds. Non blocking comment, just passing by  :slightly_smiling_face:

Comment (dofinn) [reply Makefile]: Thanks. Updated PR to include this. Much nicer then exporting CONTAINER_ENGINE via ~/.bash\|zshrc

**Comment by clyang82** [comment]
@dofinn Could you rebase your PR to pass the test? Thanks.

**Comment by dofinn** [comment]
@clyang82 https://app.circleci.com/pipelines/github/observatorium/operator/197/workflows/93d14fb0-0ee4-4b35-9210-13454231d70f/jobs/491 looks like the namespace definition may break this e2e. What is desirable from your PoV here?

File: manifests/cluster_role_binding.yaml [hunk 2]
```diff
@@ -13,4 +13,4 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: observatorium-operator
-  namespace: default
+  namespace: observatorium
```
Comment (clyang82) [inline manifests/cluster_role_binding.yaml]: For the e2e tests, we deployed the observatorium-operator in `default` namespace. the operands will be deployed in `observatorium` namespace. so the clusterRole/ClusterRoleBinding/ServiceAccount should be in `default` namespace.

Comment (dofinn) [reply manifests/cluster_role_binding.yaml]: sgtm
revered/pushed/rebased.

Comment (clyang82) [reply manifests/cluster_role_binding.yaml]: Thank you

**Pull request closed without merging**